### PR TITLE
docs(guide/replication): add a note that it's not something you're likely to need to do

### LINKED ISSUE #3140: Documentation: in "Replicating onchain state", add a callout explaining that you should never have to do this yourself and this is just an illustration
In https://mud.dev/guides/replicating-onchain-state

Additionally:
> MUD provides much of this out of the box. It uses a [familiar data model](https://mud.dev/store/data-model) with tables and fields, built on a standardized storage protocol. This lets us provide you with an [automatic indexer](https://mud.dev/services/indexer), no code is necessary to use it. And because it's all standardized, our client libraries already know how to get your app's onchain state and keep your frontend [in sync with the chain](https://mud.dev/guides/replicating-onchain-state).

Quote found in https://mud.dev/introduction. The "in sync with the chain" links to this "Guide". I think we should either link to something else (what is the best page that demonstrates "syncing to the client?", maybe it could be the top-level page of the badly named "State Query". It would explain in abstract how different parts of MUD enable clients to query the state and keep it up to date, then link to individual implementations for different programming languages)

### PR DESCRIPTION
**Pull request by qbzzt**
closes #3140

### CODE REVIEW

File: docs/pages/guides/_meta.js [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
   "extending-a-world": "Extending a World",
   "adding-delegation": "Adding Delegation",
   "modules": "Writing MUD Modules",
   emojimon: "Emojimon",
   testing: "Testing",
+  "---": {
+    title: "", // no title renders as a line
+    type: "separator",
+  },  
+  "replicating-onchain-state": "Replicating onchain state",
```
Comment (holic) [inline docs/pages/guides/_meta.js:12]: Looking at this with fresh eyes, it's really unclear why there's a separation:

image

File: docs/pages/guides/replicating-onchain-state.mdx [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
 This guide walks through how you might recreate onchain state from MUD's Store events. We use this pattern in our sync stack to hydrate a client or indexer from blockchain logs fetched from an RPC.
 
+<Callout type="info" emoji="ⓘ">
+  In most cases, you would use a pre-written [template](/templates), which includes code to synchronize information.
+  If that is not an option, you might be able to use [an indexer](/indexer) to retrieve the onchain data.
+
+This guide is mostly provided as an illustration, and to simplify the writing of future clients.
+
+</Callout>
+
```
Comment (holic) [inline docs/pages/guides/replicating-onchain-state.mdx:14]: Do we really need a callout?

```suggestion
# Replicating onchain state

This guide walks through how you might recreate onchain state from MUD's Store events. We use this pattern in our sync stack to hydrate a client or indexer from blockchain logs fetched from an RPC.

If you are using a MUD template, one of our sync packages, or indexers, all of this is already done for you. This guide is meant to demonstrate how the Store protocol works and show how you might implement a sync strategy or indexer of your own.

```

File: docs/pages/templates.mdx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Templates
+
+MUD templates include client code as well as an onchain MUD `World`.
+These integrations include data synchronization, which gives the client a read-only copy of the data in the [MUD tables](/store/tables).
+
+## Lattice templates
```
Comment (holic) [inline docs/pages/templates.mdx:6]:
```suggestion
## Official templates
```

File: docs/pages/templates.mdx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+We wrote a number of integrations between TypeScript and MUD.
+
+- [Vanilla](/templates/typescript/vanilla)
+- [React using RECS](/templates/typescript/react-ecs)
+- React using Zustand
+- [Three.js](/templates/typescript/threejs)
+
+## Third party templates
```
Comment (holic) [inline docs/pages/templates.mdx:15]:
```suggestion
## Community templates
```

File: docs/pages/guides/replicating-onchain-state.mdx (hunk 2 above)
Comment (qbzzt) [reply docs/pages/guides/replicating-onchain-state.mdx:14]: @ludns , I don't feel strongly either way - can you and @holic discuss?

File: docs/pages/guides/_meta.js [hunk 5]
```diff
@@ -1,9 +1,13 @@
...
   "hello-world": "Hello World",
   "extending-a-world": "Extending a World",
   "adding-delegation": "Adding Delegation",
   "modules": "Writing MUD Modules",
   emojimon: "Emojimon",
   testing: "Testing",
+  "---": {
+    title: "", // no title renders as a line
+    type: "separator",
+  },
```
Comment (ludns) [inline docs/pages/guides/_meta.js:11]:
```suggestion
```

File: docs/pages/guides/replicating-onchain-state.mdx [hunk 6]
```diff
@@ -1,7 +1,11 @@
+import { Callout } from "nextra/components";
+
```
Comment (ludns) [inline docs/pages/guides/replicating-onchain-state.mdx:2]:
```suggestion
```

**Pull request merged**